### PR TITLE
Adds _malloc and _free exports and e2e tests them

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This adds exports to replace ones discouraged by TinyGo. To ensure these work, this uses wazero to test them.

### CODE REVIEW

File: exports/exports.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Package exports contains WebAssembly function exports for memory allocation.
+// It is not necessary to import these otherwise.
+//
+// Ex.
+//
+//	import _ github.com/tetratelabs/tinymem/exports
```
Comment (codefromthecrypt) [inline exports/exports.go:6]: ps this actually works.. so basically if writing for tinygo, just `_` similar to go:embed!

**Comment by codefromthecrypt** [comment]
PS I don't really care about the export names, just we shouldn't conflict with TinyGo "malloc" and "free" if anyone has suggestions, let's change them!

File: exports/exports.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+//
+// Note: This is an ownership transfer, which means the caller must call free
+// when finished.
+//
+//export _malloc
+func malloc(size uint32) uintptr {
+	buf := make([]byte, size)
+	ptr := &buf[0]
+	unsafePtr := uintptr(unsafe.Pointer(ptr))
+	alivePointers[unsafePtr] = buf
```
Comment (jcchavezs) [inline exports/exports.go:24]: naive question: shall this check if the key already exists and either avoid the replacement or free the existing and accept the override?

File: exports/exports.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
+// Package exports contains WebAssembly function exports for memory allocation.
+// It is not necessary to import these otherwise.
+//
+// Ex.
+//
+//	import _ github.com/tetratelabs/tinymem/exports
+package exports
```
Comment (anuraaga) [inline exports/exports.go:7]: Any thoughts on putting in the main package? It reminds me of `_` import when embedding I be to `[]byte` and not when using `embed.FS`.

But maybe too mysterious, don't feel strongly but just an idea.

**Review by anuraaga — APPROVED** [review]

File: exports/exports_test.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+func Test_mallocRoundTrip(t *testing.T) {
+	t.Run("free when not yet allocated", func(t *testing.T) {
+		free(123) // doesn't panic
+	})
+
+	size := uint32(3)
+
+	ptr := malloc(3)
```
Comment (anuraaga) [inline exports/exports_test.go:18]:
```suggestion
	ptr := malloc(size)
```

**Review by mathetake — APPROVED** [review]

**Pull request merged**